### ISSUE TITLE
Umbraco v12 (RC1) has a breaking-change

### ISSUE THREAD

**Issue by skttl** [leekelleher/umbraco-contentment#322]
### Which Contentment version are you using?

4.4.2

### Which Umbraco version are you using? For example: 8.14.1 - don't just write v8

12.0.0-rc1

### Bug summary

The site fails to boot, giving this message:

```
An error occurred while starting the application.
TypeLoadException: Method 'GetDeliveryApiValue' in type 'Umbraco.Community.Contentment.Web.PublishedCache.DetachedPublishedProperty' from assembly 'Umbraco.Community.Contentment, Version=4.4.2.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
Unknown location

ReflectionTypeLoadException: Could not load all types from "Umbraco.Community.Contentment, Version=4.4.2.0, Culture=neutral, PublicKeyToken=null" due to LoaderExceptions, skipping:
. System.TypeLoadException on Umbraco.Community.Contentment.Web.PublishedCache.DetachedPublishedProperty: Method 'GetDeliveryApiValue' in type 'Umbraco.Community.Contentment.Web.PublishedCache.DetachedPublishedProperty' from assembly 'Umbraco.Community.Contentment, Version=4.4.2.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.

Method 'GetDeliveryApiValue' in type 'Umbraco.Community.Contentment.Web.PublishedCache.DetachedPublishedProperty' from assembly 'Umbraco.Community.Contentment, Version=4.4.2.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
Umbraco.Cms.Core.Composing.TypeFinder.GetTypesWithFormattedException(Assembly a)

Loader Exceptions:
Method 'GetDeliveryApiValue' in type 'Umbraco.Community.Contentment.Web.PublishedCache.DetachedPublishedProperty' from assembly 'Umbraco.Community.Contentment, Version=4.4.2.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
TypeLoadException: Method 'GetDeliveryApiValue' in type 'Umbraco.Community.Contentment.Web.PublishedCache.DetachedPublishedProperty' from assembly 'Umbraco.Community.Contentment, Version=4.4.2.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
```

### Steps to reproduce

Attempt to upgrade an existing site using Contentment from Umbraco 11 to Umbraco 12 rc1

### Expected result / actual result

Not sure if I expected it to just work - but obviously theres a breaking change in Umbraco, that needs handling in Contentment :)

### Do you have Umbraco ModelsBuilder enabled?

- [X] Yes, it is enabled.

### What browsers are you seeing the problem on?

Microsoft Edge

**Comment by skttl** [comment]
This looks like a "fun" project to fix. I guess v12 is going to be .net 7 too, so you can't multitarget the same way.

**Comment by leekelleher** [comment]
Fixed in [Contentment v4.4.4](https://github.com/leekelleher/umbraco-contentment/releases/tag/4.4.4). Turns out the resolution was a tidy code refactor. Thanks again for reporting this, I doubt I would have got around to it myself until the weekend.
